From d38261a20f318ce0dce1c54c6659b092a860f793 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Mon, 22 Jan 2024 16:21:29 +0200
Subject: [PATCH] [SW-0] gaudi3: pif: don't jump from gen1 to gen5

the default configuration is to allow direct jump from gen1 to gen5.
testing show that when such jump happens on HLS3 the link is
unstable. to improve stability, this option is disabled, and the link
should speedup gradually

Change-Id: I6ed8e49bb4073403d43522d5217eb8fc3d952b36
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 97bae5d..7fbc47a 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -1030,6 +1030,12 @@
 
 	/* drop Vendor Defined Messages (VDM) type 1 */
 	regmap_writel(0, D0_PCIE_DBI_SIG_REG(mmPCIE_DBI_FILTER_MASK_2_OFF));
+
+	/* prevent jumping from gen1 to gen5 directly */
+	regmap_rmwritel(D0_PCIE_DBI_SIG_REG(mmPCIE_DBI_PL32G_CONTROL_REG), 1,
+			PCIE_DBI_PL32G_CONTROL_REG_EQ_BYPASS_HIGHEST_RATE_DISABLE_M);
+	regmap_rmwritel(D0_PCIE_DBI_SIG_REG(mmPCIE_DBI_PL32G_CAPABILITY_REG), 0,
+			PCIE_DBI_PL32G_CAPABILITY_REG_EQ_BYPASS_HIGHEST_RATE_SUPPORT_M);
 }
 
 __weak void preboot_update_bar_sizes(void) {}
